go/multipooler/init.go

```diff
@@ -94,29 +97,62 @@ func Init() {
 		logger.Error("table group is required")
 		os.Exit(1)
 	}
-
-	// Register multipooler gRPC service with servenv's GRPCServer
-	if servenv.GRPCCheckServiceMap("pooler") {
-		poolerServer = server.NewMultiPoolerServer(logger, &server.Config{
-			SocketFilePath: socketFilePath,
-			PoolerDir:      poolerDir,
-			PgPort:         pgPort,
-			Database:       database,
-		})
-		poolerServer.RegisterWithGRPCServer(servenv.GRPCServer)
-		logger.Info("MultiPooler gRPC service registered with servenv")
-	}
-
 	// Create MultiPooler instance for topo registration
 	multipooler := topo.NewMultiPooler(serviceID, cell, servenv.Hostname, tableGroup)
 	multipooler.PortMap["grpc"] = int32(servenv.GRPCPort())
 	multipooler.PortMap["http"] = int32(servenv.HTTPPort())
 	multipooler.Database = database
-
-	tr = toporeg.Register(
-		func(ctx context.Context) error { return ts.RegisterMultiPooler(ctx, multipooler, true) },
-		func(ctx context.Context) error { return ts.UnregisterMultiPooler(ctx, multipooler.Id) },
-		func(s string) { serverStatus.InitError = s },
+	multipooler.ServingStatus = clustermetadatapb.PoolerServingStatus_NOT_SERVING
+
+	// Initialize the MultiPoolerManager (following Vitess tm_init.go pattern)
+	logger.Info("Initializing MultiPoolerManager")
+	poolerManager := manager.NewMultiPoolerManager(logger, &manager.Config{
+		SocketFilePath: socketFilePath,
+		PoolerDir:      poolerDir,
+		PgPort:         pgPort,
+		Database:       database,
+		TopoClient:     ts,
+		ServiceID:      multipooler.Id,
+	})
+
+	// Start the MultiPoolerManager
+	poolerManager.Start()
+
+	servenv.OnRun(
```

Comment:
What was the reason for moving this from main.go to here?

Reply:
In `main`, it was being called like this:

```go
servenv.OnRun(multipooler.Init)
```

This caused an issue because `poolerManager.Start()` also registers its own `servenv.OnRun`, resulting in a recursive call to `OnRun`.

I refactored it to just use `init`, which internally calls `start` for the different components. I have a follow-up PR planned to remove this `servenv.OnRun` entirely and register the gRPC service the same way we’re doing it for the manager now.

This matches the pattern we used in Vitess. I think it’s cleaner. What do you think?